tinymce-basics/customizing-the-table-toolbar.html

```diff
@@ -0,0 +1,583 @@
+            // to use the content_css option to load a separate CSS file.
+            // https://www.tiny.cloud/docs/configure/content-appearance/#content_style
+            // https://www.tiny.cloud/docs/configure/content-appearance/#content_css
+            content_style: `
+                table {
+                    width: 100%;
+                }
+
+                /* Add a gray background to table headers */
+                table th {
+                    background-color: #efefef;
+                    text-align: left;
+                }
+            `,
+
+            // Make the custom icon pack available in TinyMCE. A custom icon pack
+            // extends the default icon pack.
+            // https://www.tiny.cloud/docs/configure/editor-appearance/#icons
+            icons: 'customIcons',
+
+            // In this demo we don't want to use the right click context menus
+            // https://www.tiny.cloud/docs/configure/editor-appearance/#contextmenu
+            contextmenu: false,
+
+            // Tell TinyMCE how to modify the table HTML when converting regular rows
+            // to header rows. In this case td's will move into `thead` and be
+            // converted to `th`.
+            // https://www.tiny.cloud/docs/plugins/table/#table_header_type
+            table_header_type: 'sectionCells',
+
+            // Configure the table context toolbar. The last button is
+            // the built in remove table button.
+            // https://www.tiny.cloud/docs/plugins/table/#table_toolbar
+            table_toolbar: 'celloptions rowoptions columnoptions | tabledelete',
+
+            setup: (editor) => {
+
+                // Register custom keyboard shortcuts for adding rows and columns
+                // Since both the shortcut method and the menuitem method share the
+                // same actions, the actions are extracted and put into a global
+                // object. The keycode numbers 37 to 40 represent the arrow keys.
+                // For a list of javascript keycodes see
+                // https://css-tricks.com/snippets/javascript/javascript-keycodes/
+                // https://www.tiny.cloud/docs/advanced/keyboard-shortcuts/
+                editor.addShortcut('alt+37', "Insert column before", () => {
+                    menuActionManager.insertColumnBefore();
+                });
+
+                editor.addShortcut('alt+39', "Insert column after", () => {
+                    menuActionManager.insertColumnAfter();
+                });
+
+                editor.addShortcut('alt+38', "Insert row above", () => {
+                    menuActionManager.insertRowBefore();
+                });
+
+                editor.addShortcut('alt+40', "Insert row after", () => {
+                    menuActionManager.insertRowAfter();
+                });
+
+                editor.addShortcut('ctrl+meta+M', "Merge selected cells", () => {
```

Comment:
This really doesn't make sense as a shortcut. It works on Mac, but on Windows/Linux I'm not sure it'd ever work.

Reply:
Haha you're right! I went through both Google Docs and Word Online hunting for table shortcuts and those two had surprisingly few, like none really. So I ended up in the manual for Apple Pages (the native desktop app) which had shortcuts for most actions, but yeah, some of them doesn't make sense on other OSes 🤦‍♂️. I'll just remove this one and the other one.